### ISSUE TITLE
Issue with creating topics and manuals with large name length

### ISSUE THREAD

**Issue by MattJohnstone** [splunk/ponydocs#44]
I'm running into a couple of issues here: one where I define a manual with a rather large manualShortName, and one where I create a topic with a rather large topic title/display name.

The first issue comes when modifying the /Documentation:productShortName:Manuals page to define a manual with a large manualShortName. When clicking on the link to create the TOC for said manual, the topics added to the TOC (regardless of topic title size) are not being auto-generated. The topics on the TOC aren't even being shown as hyperlinks, they appear to be plaintext. Going back to the manual list page, it will still say right below the manual link "Click manual to create TOC for versionNumber." as if I never created one.

Whilst having a rather large manualShortName wouldn't make much sense to a human being, there's the possibility that the manuals are being automatically created by a script or bot.

The second issue comes after defining a manual as one normally would (using a small-in-size manualShortName). When modifying the TOC, links are supposed to be auto-generated for each topic that gets added. If a topic title is too large, the link to said topic always refers the user to a "The Requested Topic Does Not Exist" page. Even if I were to click "create page", I still get the same error upon clicking the link.

The second issue is actually something I'm having big trouble with. In my situation, the topics ARE being automatically added to the TOC, and having a large topic title is a real possibility that I do not have control over.

**Comment by firebus** [comment]
Interesting. Any idea how long the Manual or Topic title needs to be before things get borked?

Sounds like we need to first document the limits, next validate them so that you don't get these silent failures, and then finally look for a way to remove the limits...

**Comment by MattJohnstone** [comment]
Just did a couple more tests to find those.

I've found that the longer the manual name, the shorter the topic name has to be before running into the "The Requested Topic Does Not Exist" page. I haven't tested it, but I'm going to take a guess and say that the product name and version name are also a factor since those are also being stored in the page name string.

In what format is the entire page name being stored? If it's the length of the entire page name that's the issue, how can we go about increasing the limit so we can prevent this issue from happening again?

**Comment by firebus** [comment]
Yes - the page_title for a topic is "Product:Manual:TopicTitle:Version", and it all gets stored in the page_title field of the page table which, you guessed it, is VARBINARY(255).

So that's a limit.

I wonder if that limit is increased in the current version of MW, which we're working on upgrading to support: nope. Still 255 per https://www.mediawiki.org/wiki/Manual:Page_table

I can't find any discussion on the internet about long page titles in MW. You could try altering the table to change the type to TEXT or something else larger. I don't know if there's anything else in MW that expects or relies on a 255 character limit. You might also try asking in one of the MW IRC channels: https://www.mediawiki.org/wiki/MediaWiki_on_IRC

**Comment by MattJohnstone** [comment]
Wouldn't the full page_title be "Documentation:Product:Manual:TopicTitle:Version"? When I attempt to create a page manually, Mediawiki actually prevents me from making one where the title exceeds 255 characters. Any title less than or equal to 255 characters and I don't run into the issue. This is expected.
Using Ponydocs-generated pages, I'm running into the issue when the full page_title (using the format I mentioned above) exceeds not only 255 characters, but a mere 70 characters.

There seems to be three different cases:
1) If the full page_title is less than 70 characters, no issues come up.
2) If the full page_title is more than 70 characters, but the last colon before the version name is before or on the 70th character, the page will get created but can't be referred to. Any links that refer to it will just bring the user to a blank page with no text. If one were to attempt to edit the page and look at the URL, they will find that the version name is either partially cut off or missing entirely. Modifying the URL on the edit page to include the full version name will bring the user to an edit page that includes the content that the page should be providing.
3) If the full page_title is more than 70 characters and the last colon before the version name is after the 70th character, any links to the page will result in "The Requested Topic Does Not Exist". Even attempting to add content to the page will result in the error. Looking at the URL on the edit page shows the "ugly" URL formatting, but no information is cut off. If one were to modify the URL on the edit page to the "pretty" URL format instead, the edit page will include the content that the page should be providing.

**Comment by firebus** [comment]
No, I'm pretty sure the Documentation: is not part of the page_title - it's the Namespace, so it's there in the URL but not in the page_title.
